### PR TITLE
Ensure ServerAddress#sameHost parses IPV6 addresses correctly

### PR DESCRIPTION
**Pull request by rozza**
JAVA-2593

### CODE REVIEW

File: driver-core/src/main/com/mongodb/ServerAddress.java [hunk 1]
```diff
@@ -219,14 +219,7 @@ public static int defaultPort() {
...
-        String hostToUse = hostName;
-        int idx = hostToUse.indexOf(":");
-        int portToUse = defaultPort();
-        if (idx > 0) {
-            portToUse = Integer.parseInt(hostToUse.substring(idx + 1));
-            hostToUse = hostToUse.substring(0, idx);
-        }
-
-        return getPort() == portToUse && getHost().equalsIgnoreCase(hostToUse);
+        ServerAddress alternativeAddress = new ServerAddress(hostName);
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/ServerAddress.java]: How about just:

            return equals(new ServerAddress(hostName));

And given that, should we just deprecate this method entirely?

Comment (rozza) [reply driver-core/src/main/com/mongodb/ServerAddress.java]: Agreed

**Comment by rozza** [comment]
Updated

File: driver-core/src/main/com/mongodb/ServerAddress.java [hunk 2]
```diff
@@ -217,16 +217,9 @@ public static int defaultPort() {
      *
      * @param hostName the address to compare
      * @return if they are the same
+     * @Deprecated use the {@link #equals(Object)} method instead
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/ServerAddress.java]: The Javadoc tag is lower-cased.  Also add an @ Deprecated annotation.

**Pull request closed without merging**